project/ws/app/src/lib/routes/browse-by-provider/services/content-read-resolver.service.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable } from '@angular/core'
+import { ActivatedRouteSnapshot, Resolve, RouterStateSnapshot } from '@angular/router'
+import { IResolveResponse } from '@sunbird-cb/utils'
+import { Observable, of } from 'rxjs'
+import { catchError, map, tap } from 'rxjs/operators'
+import { WidgetContentService } from '@sunbird-cb/collection/src/lib/_services/widget-content.service'
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ContentReadResolverService implements
+  Resolve<
+  Observable<IResolveResponse<any>> | IResolveResponse<any>
```

Comment:
Is this is fine?

Reply:
ya this is fine because we are using resolver